backend-shared/config/db/util/utils_test.go

```diff
@@ -1,12 +1,321 @@
+
+			dbQueries, err = db.NewUnsafePostgresDBQueries(false, false)
+			Expect(err).To(BeNil())
+
+			// ----------------------------------------------------------------------------
+			By("First create resources required by deploymentToApplicationMapping.")
+			// ----------------------------------------------------------------------------
+
+			// Create ClusterCredentials
+			clusterCredentials = db.ClusterCredentials{
+				Host:                        "host",
+				Kube_config:                 "kube-config",
+				Kube_config_context:         "kube-config-context",
+				Serviceaccount_bearer_token: "serviceaccount_bearer_token",
+				Serviceaccount_ns:           "Serviceaccount_ns",
+			}
+			err = dbQueries.CreateClusterCredentials(ctx, &clusterCredentials)
+			Expect(err).To(BeNil())
+
+			// ---------------------------------------------------------------------------
+			// Create ManagedEnvironment
+
+			managedEnvironment = db.ManagedEnvironment{
+				Clustercredentials_id: clusterCredentials.Clustercredentials_cred_id,
+				Name:                  "my-managed-environment",
+			}
+			err = dbQueries.CreateManagedEnvironment(ctx, &managedEnvironment)
+			Expect(err).To(BeNil())
+
+			// ---------------------------------------------------------------------------
+			// Create GitopsEngineCluster
+
+			gitopsEngineCluster = db.GitopsEngineCluster{
+				Clustercredentials_id: clusterCredentials.Clustercredentials_cred_id,
+			}
+			err = dbQueries.CreateGitopsEngineCluster(ctx, &gitopsEngineCluster)
+			Expect(err).To(BeNil())
+
+			// ---------------------------------------------------------------------------
+			// Create GitopsEngineInstance
+
+			gitopsEngineInstance = db.GitopsEngineInstance{
+				Namespace_name:   "my-namespace",
+				Namespace_uid:    "test-1",
+				EngineCluster_id: gitopsEngineCluster.Gitopsenginecluster_id,
+			}
+			err = dbQueries.CreateGitopsEngineInstance(ctx, &gitopsEngineInstance)
+			Expect(err).To(BeNil())
+
+			// ---------------------------------------------------------------------------
+			// Create Application
+
+			application = db.Application{
+				Name:                    "my-application",
+				Spec_field:              "{}",
+				Engine_instance_inst_id: gitopsEngineInstance.Gitopsengineinstance_id,
+				Managed_environment_id:  managedEnvironment.Managedenvironment_id,
+			}
+			err = dbQueries.CreateApplication(ctx, &application)
+			Expect(err).To(BeNil())
+
+			// ---------------------------------------------------------------------------
+			// Create deploymentToApplicationMapping request data
+
+			deploymentToApplicationMapping = &db.DeploymentToApplicationMapping{
+				Deploymenttoapplicationmapping_uid_id: string(uuid.NewUUID()),
+				Application_id:                        application.Application_id,
+				DeploymentName:                        "my-depl-to-app-mapping",
+				DeploymentNamespace:                   "my-namespace",
+				NamespaceUID:                          "test-1",
+			}
```

Comment:
For the above function, can't we use CreateSampleData functionality for the required components, rest can be created as above?

Reply:
Yes, it could be done, my thinking here was that it Is is not repeated anywhere, just used at one place. If needed I am Okay to move it into a function. WDYT?

Reply:
Hmm, so what I meant was using CreateSampleData for creating *ClusterCredentials, *ManagedEnvironment, *GitopsEngineCluster, and *GitopsEngineInstance from the existing function, the rest of the code can be kept as similar to what you have written? Is that what you were saying or did I miss something?